judge/views/two_factor.py

```diff
@@ -0,0 +1,236 @@
+import base64
+import json
+import os
+from io import BytesIO
+
+import pyotp
+import qrcode
+import webauthn
+from django.conf import settings
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.contrib.auth.views import SuccessURLAllowedHostsMixin
+from django.http import Http404, HttpResponse, HttpResponseBadRequest, HttpResponseRedirect, JsonResponse
+from django.urls import reverse
+from django.utils.http import is_safe_url
+from django.utils.translation import gettext as _
+from django.views.generic import FormView, View
+from django.views.generic.detail import SingleObjectMixin
+
+from judge.forms import TOTPForm, TwoFactorLoginForm
+from judge.jinja2.gravatar import gravatar
+from judge.models import WebAuthnCredential
+from judge.utils.two_factor import WebAuthnJSONEncoder, webauthn_encode
+from judge.utils.views import TitleMixin
+
+
+class TOTPView(TitleMixin, LoginRequiredMixin, FormView):
+    form_class = TOTPForm
+
+    def get_form_kwargs(self):
+        result = super(TOTPView, self).get_form_kwargs()
+        result['profile'] = self.profile
+        return result
+
+    def dispatch(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
+            self.profile = self.request.profile
+            if self.check_skip():
+                return self.next_page()
+        return super(TOTPView, self).dispatch(request, *args, **kwargs)
+
+    def check_skip(self):
+        raise NotImplementedError()
+
+    def next_page(self):
+        return HttpResponseRedirect(reverse('user_edit_profile'))
+
+
+class TOTPEnableView(TOTPView):
+    title = _('Enable Two Factor Authentication')
```

Comment:
```suggestion
    title = _('Enable Two-Factor Authentication')
```

Reply:
`Two-Factor` has incorrect capitalization. Also this is out of scope because it was actually from `judge/views/totp.py`. Somehow it was detected as a new file instead of rename.